sql-ballerina/Module.md

```diff
@@ -0,0 +1,302 @@
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<record{}, sql:Error> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(record{} student) {
+    //Can perform any operations using the 'student' and can access any fields in the returned record.
+});
+```
+
+There are situations in which you may not want to iterate through the database and in that case, you may decide
+to only use the `next()` operation in the result `stream` and retrieve the first record. In such cases, the returned
+result stream will not be closed, and you have to explicitly invoke the `close` operation on the 
+`sql:Client` to release the connection resources and avoid a connection leak as shown below.
+
+```ballerina
+stream<record{}, sql:Error> resultStream = 
+            dbClient->query("SELECT count(*) as total FROM students");
+
+record {|record {} value;|}|error? result = resultStream.next();
+
+if result is record {|record {} value;|} {
+    // A valid result is returned.
+} else if result is error {
+    // An error is returned as the result.
+} else {
+    // Student table must be empty.
+}
+
+error? e = resultStream.close();
+```
+
+#### Updating Data
+
+This example demonstrates modifying data by executing an `UPDATE` statement via the `execute` remote function of 
+the client.
+
+```ballerina
+int age = 23;
+sql:ParameterizedQuery query = `UPDATE students SET name = 'John' 
+                                WHERE age = ${age}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Deleting Data
+
+This example demonstrates deleting data by executing a `DELETE` statement via the `execute` remote function of 
+the client.
+
+```ballerina
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This example demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the 
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and 
+parameterized SQL query as same as the above `execute` operations.
+
+```ballerina
+// Create the table with the records that need to be inserted.
+var data = [
+  { name: "John", age: 25  },
+  { name: "Peter", age: 24 },
+  { name: "jane", age: 22 }
+];
+
+// Do the batch update by passing the batches.
+sql:ParameterizedQuery[] batch = from var row in data
+                                 select `INSERT INTO students ('name', 'age')
+                                 VALUES (${row.name}, ${row.age})`;
+sql:ExecutionResult[] ret = check dbClient->batchExecute(batch);
+```
+
+#### Execute SQL Stored Procedures
```

Comment:
@lahirusamith I think we have to improve these examples to include on how to process input params return multiple query sets. 

IMO whatever that is in the be should come here and we should simply the be. 

@kalaiyarasiganeshalingam @daneshk WDYT?

Reply:
Do you mean having inout parameter in the example?
https://dev.ballerina.io/learn/by-example/mysql-call-stored-procedures.html

Reply:
Yes. +1 to improve it in the next iteration too